src/commands/deploy.ts

```diff
@@ -80,18 +104,32 @@ export default class Deploy extends Command {
    */
   public async isInteractive(interactive: boolean): Promise<boolean> {
     if (interactive) return true;
-    const deployFileExists = await fs.fileExists(DEPLOY_OPTIONS_FILE);
+    const deployFileExists = await exists(DEPLOY_OPTIONS_FILE);
     return deployFileExists ? false : true;
   }
 
   public async readOptions(): Promise<Record<string, Deployer.Options>> {
-    if (await fs.fileExists(DEPLOY_OPTIONS_FILE)) {
-      return (await fs.readJson(DEPLOY_OPTIONS_FILE)) as Record<string, Deployer.Options>;
+    if (await exists(DEPLOY_OPTIONS_FILE)) {
+      return readJson<Record<string, Deployer.Options>>(DEPLOY_OPTIONS_FILE);
     } else {
       return {};
     }
   }
 
+  public async commit(): Promise<void> {
+    const gitignore = await readFile('.gitignore', 'utf-8');
+    if (!gitignore.includes(DEPLOY_OPTIONS_FILE)) {
+      const addition = `${EOL}${EOL}# Deploy Options${EOL}${DEPLOY_OPTIONS_FILE}${EOL}`;
+      await writeFile('.gitignore', `${gitignore}${addition}`);
+    }
+    exec('git add .gitignore', { silent: true });
+    exec(`git commit -am "Add ${DEPLOY_OPTIONS_FILE} to .gitignore"`, { silent: true });
```

Comment:
I don't think running `git` commands for the user is good UX. Simply adding the file to `.gitignore` will cause git to ignore it and we can leave it up to the user to commit it to their repo if they wish.

I also think we should output a message that says something like "Adding deploy-options.json to .gitignore". We could even add something like "Run `git add` and `git commit` to commit it to your repo".

Reply:
Resolved in design time conversation.